### PR TITLE
asim/fix-network-parsres-post-data-test

### PR DESCRIPTION
**Pull request by oshezaf**

### CODE REVIEW

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoft365Defender.yaml [hunk 1]
```diff
@@ -47,33 +49,37 @@ ParserQuery: |
...
       UrlHostname = SplitUrl[0],
-      UrlDomain = strcat_array(array_slice(SplitUrl, 1, -1), '.'),
-      SrcDomainType = "FQDN",
-      DvcDomainType = "FQDN",
-      DstDomainType = "FQDN",
+      UrlDomain = tostring(strcat_array(array_slice(SplitUrl, 1, -1), '.')),
+      UrlFQDN = iif(RemoteUrl contains ".", RemoteUrl, "")
+    | extend
+      DvcDomainType = iif(DvcFQDN != "", "FQDN", ""),
+      UrlDomainType = iff(UrlFQDN != "", "FQDN", ""),
```
Comment (YaronFruchtmann) [inline Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoft365Defender.yaml:73]: I am not sure the method for parsing Host works for Url. for example contoso.com -  Url domain would be "com"

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoft365Defender.yaml [hunk 2]
```diff
@@ -109,37 +111,41 @@ ParserQuery: |
       EventType = 'NetworkSession',
       EventSeverity = "Informational",
       DvcIdType = 'MDEid'
-    | project-rename
-      NetworkProtocol = Protocol,
-      DvcFQDN = DeviceName
+    | extend
+      RemoteUrl = extract (@"(?:https?://)?(.*)", 1, RemoteUrl)
```
Comment (YaronFruchtmann) [inline Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoft365Defender.yaml:115]: no risk of RemoteUrl including port, query, fragments?

**Review by YaronFruchtmann — COMMENTED** [review]
version number changed?

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoft365Defender.yaml (hunk 1 above)
Comment (oshezaf) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoft365Defender.yaml:73]: first com is a domain.... there is no easy way around this. Essentially you have to do a lookup for known TLDs. I will leave it as is.

**Comment by oshezaf** [comment]
> version number changed?

Will change

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoft365Defender.yaml (hunk 1 above)
Comment (yaronMSFT) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoft365Defender.yaml:73]: ok. maybe we need to push this as a kusto request, more sophisticated and accurate extraction of the domain?

**Review by YaronFruchtmann — APPROVED** [review]

**Comment by oshezaf** [comment]
@YaronFruchtmann : I did remove the extra parameters. I need a fast review here. Sorry for the additional trouble.

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoft365Defender.yaml [hunk 3]
```diff
@@ -28,11 +28,13 @@ ParserQuery: |
     ,'ConnectionRequest','Outbound', true
     ,'InboundConnectionAccepted', 'Inbound', false
     ,'ConnectionFound', 'Unknown', false
-    ,'ListeningConnectionCreated', 'Listening', false 
+    ,'ListeningConnectionCreated', 'Listen', false
```
Comment (YaronFruchtmann) [inline Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoft365Defender.yaml:31]: Version update missing

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionzScalerZIA.yaml [hunk 4]
```diff
@@ -33,7 +33,10 @@ ParserQuery: |
     'Block/ICMP','Drop ICMP',
     'Block/Reset', 'Reset',
     'IPS Drop', 'Drop',
-    'IPS Reset', 'Reset'
+    'IPS Reset', 'Reset',
```
Comment (YaronFruchtmann) [inline Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionzScalerZIA.yaml:36]: version

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoft365Defender.yaml [hunk 5]
```diff
@@ -32,12 +32,6 @@ ParserParams:
   - Name: dstportnumber
```
Comment (YaronFruchtmann) [inline Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoft365Defender.yaml:32]: version

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftLinuxSysmon.yaml [hunk 6]
```diff
@@ -32,12 +32,6 @@ ParserParams:
   - Name: dstportnumber
     Type: int
     Default: int(null)
-  - Name: url_has_any
-    Type: dynamic
-    Default: dynamic([])
-  - Name: httpuseragent_has_any
-    Type: dynamic 
-    Default: dynamic([])
   - Name: hostname_has_any
```
Comment (YaronFruchtmann) [inline Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftLinuxSysmon.yaml:35]: version

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftWindowsEventFirewall.yaml [hunk 7]
```diff
@@ -33,12 +33,6 @@ ParserParams:
   - Name: dstportnumber
```
Comment (YaronFruchtmann) [inline Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftWindowsEventFirewall.yaml:33]: version

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionzScalerZIA.yaml [hunk 8]
```diff
@@ -57,7 +57,10 @@ ParserQuery: |
     'Block/ICMP','Drop ICMP',
     'Block/Reset', 'Reset',
     'IPS Drop', 'Drop',
-    'IPS Reset', 'Reset'
+    'IPS Reset', 'Reset',
```
Comment (YaronFruchtmann) [inline Parsers/ASimNetworkSession/Parsers/vimNetworkSessionzScalerZIA.yaml:60]: version

**Review by YaronFruchtmann — APPROVED** [review]
I marked the ones where I think you missed updating the version number. 
Other than that - approved

**Pull request merged**